torch/_dynamo/resume_execution.py

```diff
@@ -432,7 +434,16 @@ def update(instructions: List[Instruction], code_options: Dict[str, Any]):
             # because the line number table monotonically increases from co_firstlineno
             # remove starts_line for any instructions before the graph break instruction
             # this will ensure the instructions after the break have the correct line numbers
-            target_ind = int(target.offset / 2)
+            if is_py311_plus:
+                # In 3.11+ bytecode instruction size vary, so do bisect to find the index
+                target_ind = next(
+                    idx
+                    for (idx, inst) in enumerate(instructions)
+                    if inst.offset is not None and inst.offset >= target.offset
+                )
+            else:
+                target_ind = target.offset // 2
+
             for inst in instructions[0:target_ind]:
                 inst.starts_line = None
```

Comment:
Is it guaranteed that `target.offset` would be present there? (because the reason I had to abandon bisect is that in some cases there were None's in the list)